docs/general/naming.md

```diff
@@ -0,0 +1,312 @@
+
+- Namespaces can be nested. For example `telemetry.sdk` is a namespace inside
+  top-level `telemetry` namespace and `telemetry.sdk.name` is an attribute
+  inside `telemetry.sdk` namespace.
+
+  Use namespaces (and dot separator) whenever it makes sense. For example
+  when introducing an attribute representing a property of some object, follow
+  `*{object}.{property}` pattern. Avoid using underscore (`*{object}_{property}`)
+  if this object could have other properties.
+
+- For each multi-word dot-delimited component of the name separate the
+  words by underscores (i.e. use snake_case). For example
+  `http.response.status_code` denotes the status code in the http namespace.
+
+  Use underscore only when using dot (namespacing) does not make sense or changes
+  the semantical meaning of the name. For example, use `rate_limiting`
+  instead of `rate.limiting`.
+
+- Be precise. Attribute, event, metric, and other names should be descriptive and
+  unambiguous.
+  - When introducing a name describing certain property of the object,
+    include the property name. For example, use `file.owner.name` instead of `file.owner`
+    and `system.network.packets.dropped` instead of `system.network.dropped`
+  - Avoid introducing names and namespaces that would mean different things when
+    used by different conventions or instrumentations. For example, use `security_rule`
+    instead of `rule`.
+
+- Use shorter names when it does not compromise clarity. Drop namespace
+  components or words in multi-word components when they are not necessary. For example,
+  `vcs.change.id` describes pull request id as precisely as `vcs.repository.change.id` does.
+
+## Name Abbreviation Guidelines
+
+Abbreviations MAY be used when they are widely recognized and commonly used.
+
+Examples include common technical abbreviations such as `IP`, `DB`, `CPU`,
+`HTTP`, `URL`, or product names like `AWS`, `GCP`, `K8s`.
+
+Abbreviations that are commonly recognized but only within a certain domain MAY
+be used when qualified by the corresponding namespace.
+
+For example, use `container.csi.*` instead of `container.container_storage_interface`
+or `container.oci.*` instead of `container.open_container_initiative.*`
+
+Abbreviations SHOULD be avoided if they are ambiguous, for example, when they apply
+to multiple products or concepts.
+
+## Name Reuse Prohibition
+
+Two attributes, two metrics, or two events MUST NOT share the same name.
+Different entities (attribute and metric, metric and event) MAY share the same name.
+
+Attributes, metric, events, or other named entities SHOULD NOT be removed from semantic
+conventions regardless of their maturity level. When the convention is renamed or
+no longer recommended, it SHOULD be deprecated.
+
+## Recommendations for OpenTelemetry Authors
+
+- When coming up with a new semantic convention make sure to check existing
+  namespaces ([Semantic Conventions](/docs/README.md)) to see if a similar entity
```

Comment:
(trying to avoid the word entity if it makes sense 😅)
```suggestion
  namespaces ([Semantic Conventions](/docs/README.md)) to see if a similar namespace
```